Allow setting the deployment target directly in the deploy option

iocanel opened the pull request:
Currently we use `quarkus.kubernetes.deploy` to trigger deployment.
To deploy to knative or openshfit we require an addtional option: 
- `-Dquarkus.kubernetes.deployment-target=knative`
- `-Dquarkus.kubernetes.deployment-target=openshift`

This pull request enables users to just use:

-  `-Dquarkus.knative.deploy=true` instead of `-Dquarkus.kubernetes.deploy=true -Dquarkus.kubernetes.deploymet-target=knative`
-  `-Dquarkus.openshift.deploy=true` instead of `-Dquarkus.kubernetes.deploy=true -Dquarkus.kubernetes.deploymet-target=openshift`

What's interesting with this approach is that besides the obvious UX improvement, it also opens the way of using this approach for non-kubernetes deployments.

iocanel committed:
feat: support quarkus.<deployment-target>.deploy

geoand commented on docs/src/main/asciidoc/deploying-to-kubernetes.adoc:
```diff
@@ -1014,6 +1014,21 @@ OpenShift's users might want to use `oc` rather than `kubectl`:
 ----
 
+For users that prefer to keep the `application.properties` independent of the deployment platform, the deployment target can be specified directly in the deploy command by adding `-Dquarkus.kubernetes.deployment-target=openshift`
+next to `-Dquarkus.kubernetes.deploy=true`. The two properties can be collapsed into one: `-Dquarkus.openshift.deploy=true`.
```
```suggestion
in addition to `-Dquarkus.kubernetes.deploy=true`. Furthermore, Quarkus allows for collapsing the two properties into a single one by using: `-Dquarkus.openshift.deploy=true`.
```

geoand commented on docs/src/main/asciidoc/deploying-to-kubernetes.adoc:
```diff
@@ -1098,6 +1113,21 @@ By default, when no `deployment-target` is set, then only vanilla Kubernetes res
 `quarkus.kubernetes.deployment-target=kubernetes,openshift`) then the resources for all targets are generated, but only the resources
 that correspond to the *first* target are applied to the cluster (if deployment is enabled).
 
+For users that prefer to keep the `application.properties` independent of the deployment platform, the deployment target can be specified directly in the deploy command by adding `-Dquarkus.kubernetes.deployment-target=knative`
+next to `-Dquarkus.kubernetes.deploy=true`. The two properties can be collapsed into one: `-Dquarkus.knative.deploy=true`.
```
Same as above

geoand reviewed (commented):
Very nice usability improvement!

In addition to a suggestion I added, I have the following question: What happens if a users adds something like:

```
quarkus.kubernetes.deployment-target=openshift
quarkus.knative.deploy=true
```

?

iocanel commented:
> Very nice usability improvement!
> 
> In addition to a suggestion I added, I have the following question: What happens if a users adds something like:
> 
> ```
> quarkus.kubernetes.deployment-target=openshift
> quarkus.knative.deploy=true
> ```
> 
> ?

In this case `quarkus.kubernetes.deployment-target` has higher precedence. It can't be done differently or `quarkus.kubernetes.deploy` will always shadow `quarkus.kubernetes.deployment-target` rendering it useless. 
This would result in a breaking change and loss of functionality as `quarkus.kubernetes.deploymnet-target` allows for multiple values.

Updated doc.

geoand reviewed (approved):
💪🏼
